src/components/themed/WalletList.js

```diff
@@ -161,39 +160,30 @@ export function WalletList(props: Props) {
           key: walletId
         })
       } else if (wallet != null) {
-        const { enabledTokens, name, currencyCode, currencyNames } = asSafeDefaultGuiWallet(wallet)
+        const currencyConfig = account.currencyConfig[wallet.pluginId]
+        const { builtinTokens = {}, currencyInfo, customTokens = {} } = currencyConfig
+        const { currencyCode, displayName } = currencyInfo
+        const { enabledTokens, name } = asSafeDefaultGuiWallet(wallet)
 
         // Initialize wallets
-        if (checkFilterWallet({ name, currencyCode, currencyName: currencyNames[currencyCode] }, searchText, allowedCurrencyCodes, excludeCurrencyCodes)) {
+        if (checkFilterWallet({ name, currencyCode, currencyName: displayName }, searchText, allowedCurrencyCodes, excludeCurrencyCodes)) {
           walletList.push({
             id: walletId,
             fullCurrencyCode: currencyCode,
             key: `${walletId}-${currencyCode}`
           })
         }
 
-        // Old logic on getting tokens
-        const enabledNotHiddenTokens = enabledTokens.filter(token => {
-          let isVisible = true // assume we will enable token
-          const tokenIndex = customTokens.findIndex(item => item.currencyCode === token)
-          // if token is not supposed to be visible, not point in enabling it
-          if (tokenIndex > -1 && customTokens[tokenIndex].isVisible === false) isVisible = false
-          return isVisible
-        })
-
         // Initialize tokens
-        for (const tokenCode of enabledNotHiddenTokens) {
+        const allTokens = { ...customTokens, ...builtinTokens }
+        for (const tokenCode of enabledTokens) {
+          const tokenId = Object.keys(allTokens).find(id => allTokens[id].currencyCode === tokenCode)
+          if (tokenId == null) continue
+          const token = allTokens[tokenId]
```

Comment:
This is exactly what I also had in mind that we need to do here.
This is what I did in my (now irrelevant) walletPickerModal PR:
```js
   // Creating the "add token" rows for token wallets
    const allTokens = { ...builtinTokens, ...customTokens }
    Object.keys(allTokens).forEach(tokenId => {
      // Don't add a "add token" row if we already have an existing wallet of this type
      if (existingPlugins[`${pluginId}.${tokenId}`] === true) return
      const { currencyCode, displayName } = allTokens[tokenId]
      if (filter({ tokenId, pluginId }, pluginId, filterCreate)) return
      addTokenRows.unshift({ currencyCode, displayName, tokenId, pluginId, ...curriedCurrencyIcon(tokenId) })
    })
 ```